xrootd/client.go

```diff
@@ -0,0 +1,155 @@
+// Copyright 2018 The go-hep Authors.  All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package xrootd // import "go-hep.org/x/hep/xrootd"
+
+import (
+	"context"
+	"encoding/binary"
+	"fmt"
+	"io"
+	"log"
+	"net"
+	"os"
+
+	"go-hep.org/x/hep/xrootd/encoder"
+	"go-hep.org/x/hep/xrootd/streammanager"
+)
+
+var logger = log.New(os.Stderr, "xrootd: ", log.LstdFlags)
+
+// A Client to xrootd server
+type Client struct {
+	connection      *net.TCPConn
```

Comment:
`s/connection *net.TCPConn/conn net.Conn/`

`conn` is the usual name for a connection in Go.
also. using a `net.Conn` instead of the concrete underlying type will probably help us during testing (replacing a tcp connection with a `net.Pipe` for example)